src/main/java/com/kobylynskyi/graphql/codegen/scala/ScalaGraphQLCodegen.java

```diff
@@ -0,0 +1,81 @@
+package com.kobylynskyi.graphql.codegen.scala;
+
+import com.kobylynskyi.graphql.codegen.GraphQLCodegen;
+import com.kobylynskyi.graphql.codegen.MapperFactory;
+import com.kobylynskyi.graphql.codegen.model.GeneratedInformation;
+import com.kobylynskyi.graphql.codegen.model.MappingConfig;
+import com.kobylynskyi.graphql.codegen.model.definitions.ExtendedScalarTypeDefinition;
+import com.kobylynskyi.graphql.codegen.supplier.MappingConfigSupplier;
+
+import java.io.File;
+import java.util.Collection;
+import java.util.List;
+
+/**
+ * {@inheritDoc}
+ */
+public class ScalaGraphQLCodegen extends GraphQLCodegen {
+
+    private static final MapperFactory MAPPER_FACTORY = new ScalaMapperFactoryImpl();
+
+    /**
+     * {@inheritDoc}
+     */
+    public ScalaGraphQLCodegen(List<String> schemas, File outputDir, MappingConfig mappingConfig, GeneratedInformation generatedInformation) {
+        super(schemas, outputDir, mappingConfig, generatedInformation, MAPPER_FACTORY);
+    }
+
+    /**
+     * {@inheritDoc}
+     */
+    public ScalaGraphQLCodegen(String introspectionResult, File outputDir, MappingConfig mappingConfig, GeneratedInformation generatedInformation) {
+        super(introspectionResult, outputDir, mappingConfig, generatedInformation, MAPPER_FACTORY);
+    }
+
+    /**
+     * {@inheritDoc}
+     */
+    public ScalaGraphQLCodegen(List<String> schemas, String introspectionResult, File outputDir, MappingConfig mappingConfig, MappingConfigSupplier externalMappingConfigSupplier) {
+        super(schemas, introspectionResult, outputDir, mappingConfig, externalMappingConfigSupplier, MAPPER_FACTORY);
+    }
+
+    /**
+     * {@inheritDoc}
+     */
+    public ScalaGraphQLCodegen(List<String> schemas, String introspectionResult, File outputDir, MappingConfig mappingConfig, MappingConfigSupplier externalMappingConfigSupplier, GeneratedInformation generatedInformation) {
+        super(schemas, introspectionResult, outputDir, mappingConfig, externalMappingConfigSupplier, generatedInformation, MAPPER_FACTORY);
+    }
+
+    /**
+     * {@inheritDoc}
+     */
+    @Override
+    protected void initDefaultValues(MappingConfig mappingConfig) {
+        if (mappingConfig.getGenerateBuilder() == null) {
+            // functional expression
+            mappingConfig.setGenerateBuilder(false);
+        }
+        if (mappingConfig.getGenerateImmutableModels() == null) {
+            // functional expression
+            mappingConfig.setGenerateImmutableModels(true);
+        }
+        super.initDefaultValues(mappingConfig);
```

Comment:
`super.initDefaultValues(mappingConfig);`  Maybe it should be at the top?

Reply:
If I move it to the top then `super.initDefaultValues(mappingConfig)` will set values for `setGenerateImmutableModels` and `setGenerateBuilder`, but as I understand, you don't want that.

Reply:
Yes, I understand that will override the settings